experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/node.ts

```diff
@@ -70,14 +77,28 @@ export const nodeAPi: NodeApi = {
 	): value is ProxyNode<TSchema> => {
 		return getTreeNode(value)?.is(schema) ?? false;
 	},
-	parent: (node) => {
+	parent: (node: SharedTreeNode) => {
 		const treeNode = assertTreeNode(node).parentField.parent.parent;
 		if (treeNode !== undefined) {
-			return getProxyForNode(treeNode as any); // TODO: why does this get weirdly narrowed without the `any`?
+			return getProxyForNode(treeNode);
 		}
 
 		return undefined;
 	},
+	key: (node: SharedTreeNode) => {
+		const treeNode = assertTreeNode(node);
+		const parent = nodeApi.parent(node);
+		if (parent !== undefined) {
+			const parentSchema = nodeApi.schema(parent);
+			if (schemaIsFieldNode(parentSchema)) {
```

Comment:
It isn't totally clear to me why the parent being a fieldNode indicates this is a list. Could we expand on that?

Reply:
How robust will this check be to future changes?

Reply:
Good question. We have a few checks like this in the proxy API. 

EditableTree2, in general, does not guarantee that FieldNodes have any relationship to lists. However, in the proxy API, FieldNodes are always lists. The proxy API doesn't expose sequences (fields) directly, and it also doesn't expose FieldNodes directly. Lists, which it does expose, are the combination of both (an implementation detail).

I think you're right that we could factor out the various places in the proxy API that we check FieldNode-ness and assume it means list. That's a good idea for a quality/refactor/clean-up pass (could make a work item?). I'm not worried about this breaking in the meantime though. If the assumption were broken here, it would be (more fundamentally) broken in other places in the proxy API.